mkosi/config.py

```diff
@@ -1160,6 +1161,48 @@ def config_parse_minimum_version(value: Optional[str], old: Optional[GenericVers
     return max(old, new)
 
 
+def config_path_run(script: Path) -> str:
+    if not os.access(script, os.X_OK):
+        die(f"{script} is not executable")
+
+    return run([script.absolute()], stdout=subprocess.PIPE, env=os.environ).stdout
+
+
+def config_path_read(file: Path) -> str:
+    return file.read_text()
+
+
+def config_path_run_or_read(file: Path) -> str:
+    "Run the specified file and capture its output if it's executable, else read file contents"
+    if os.access(file, os.X_OK):
+        return config_path_run(file)
+
+    content = config_path_read(file)
```

Comment:
Let's not make this overly complicated
```suggestion
    content = file.read_text()
```